### DISCUSSION TITLE
Trying to achieve <1mm voxel width for DOSXYZ calculations

### DISCUSSION THREAD

**Discussion by jafrshabazz** [nrc-cnrc/EGSnrc#982] [Questions]
Hi, I am trying to run a simple dose calculation for a water phantom with very fine resolution in the z-direction to observe dose in the build up region. But whenever I try to go smaller than 1mm, it seems as if the simulation gets stuck on one bin and just continuously recycles particles in a loop. I let the program run for close to 12 hours with only a small amount of histories, with no progress. Is it just a limitation of DOSXYZ that it cant go below 1mm? Or is there a parameter that I can change in the source code to allow for it? Below are the parameters for the phantom. I have already run multiple simulations with voxels at 1mm with all the other parameters remaining the same.

![image](https://user-images.githubusercontent.com/125318593/229841602-36ed49fe-0ed1-438a-9463-ac6d4a39cd67.png)

**Comment by rtownson** [comment]
You may need to increase the maximum number of voxels, found in `egs_home/dosxyznrc/dosxyznrc_user_macros.mortran`. Change the value to be whatever is necessary, then recompile dosxyznrc.
```
REPLACE {$IMAX}    WITH {128} "Maximum number of x cells
REPLACE {$JMAX}    WITH {128} "Maximum number of y cells
REPLACE {$KMAX}    WITH {275} "Maximum number of z cells
```

If the application fails to compile, you may need to include some additional compiler flags. To do this, in the `Makefile` change the line for `EGS_EXTRA_LIBS` to instead be:
```
EGS_EXTRA_LIBS = $(FLIBS) $(BEAMLIB_EXTRA_LIBS) -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE -mcmodel=large
```

**Reply by jafrshabazz** [reply]
Thank you for your response, unfortunately, that didn't work. The problem isn't the number of voxels that I am running, but the size. I am running a simulation now with only 18100 histories with a phase-space file that has over a billion particles in it, yet it has recycled the particles in the files multiple times. It's like it gets stuck in a loop whenever I try and go below 1mm.

![image](https://user-images.githubusercontent.com/125318593/229907456-a8255671-8ca2-4512-9bf4-e574d2e9deee.png)

**Reply by rtownson** [reply]
So when you change the voxel size it is reducing the overall dimensions of the phantom? This seems like the particles are missing the geometry entirely, so it keeps asking for more particles from the phase-space because they are all missing. Try adjusting the beam angles or positioning.
